cirq/circuits/circuit_test.py

```diff
@@ -2532,3 +2532,52 @@ def test_decompose():
     assert cirq.decompose(
         cirq.Circuit.from_ops(cirq.X(a), cirq.Y(b), cirq.CZ(a, b))
     ) == [cirq.X(a), cirq.Y(b), cirq.CZ(a, b)]
+
+
+def test_power():
+    a, b = cirq.LineQubit.range(2)
+    base = cirq.Circuit.from_ops((cirq.X ** 0.5)(a), (cirq.Y ** 0.2)(b),
+                                 (cirq.CZ)(a, b))
+    cirq.testing.assert_same_circuits(base ** 0, cirq.Circuit())
+    cirq.testing.assert_same_circuits(base ** 1, base)
+    cirq.testing.assert_same_circuits(base ** 2, base + base)
+    cirq.testing.assert_same_circuits(base ** 2.5, base + base + base ** 0.5)
+    cirq.testing.assert_same_circuits(base ** 0.5, base ** 0.5)
+    cirq.testing.assert_same_circuits(base ** -1, base ** -1)
+    cirq.testing.assert_same_circuits(base ** -2, base ** -1 + base ** -1)
+    cirq.testing.assert_same_circuits(base ** -2.5,
+                                      base ** -1 + base ** -1 + base ** -0.5)
+    cirq.testing.assert_same_circuits(base ** -0.5, base ** -0.5)
+
+
+def test_power_explicit():
+    a, b = cirq.LineQubit.range(2)
+    ops = [(cirq.X ** 0.5)(a), (cirq.Y ** 0.2)(b), cirq.CZ(a, b)]
+    sqrt_ops = [(cirq.X ** 0.25)(a), (cirq.Y ** 0.1)(b), (cirq.CZ**0.5)(a, b)]
+    base = cirq.Circuit.from_ops(ops)
+    expected_ops = ops + ops + sqrt_ops
+    cirq.testing.assert_same_circuits(base ** 2.5,
+                                      cirq.Circuit.from_ops(expected_ops))
+
+    # Note that moment does not reverse order of operations in moment.
+    inverse_ops = [(cirq.CZ ** -1)(a, b), (cirq.X ** -0.5)(a),
+                   (cirq.Y ** -0.2)(b)]
+    inverse_sqrt_ops = [(cirq.CZ ** -0.5)(a, b), (cirq.X ** -0.25)(a),
+                        (cirq.Y ** -0.1)(b)]
+
+    expected_ops = inverse_ops + inverse_ops + inverse_sqrt_ops
+    cirq.testing.assert_same_circuits(base ** (-2.5),
+                                      cirq.Circuit.from_ops(expected_ops))
+
```

Comment:
Some gates don't have inverses (e.g. measurement). Check that the behavior is correct when inverting measurement (i.e. that `cirq.inverse(..., default=None)` returns `None` instead of raising an error).

Reply:
done